### ISSUE TITLE
`list[tuple[int, int]]` considered compatible with `list[tuple[int, ...]]`

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#4835]
**To Reproduce**
```python
# pyright: strict

from typing import List, Tuple

def fun(arg: List[Tuple[int, ...]]):
    arg.append((1,2,3,4,5))

mylist: List[Tuple[int, int]] = [(1, 2)]

fun(mylist) # no errors, injects 5-tuple into mylist in runtime
```

**Expected behavior**
An error at the call site of `fun`. `mypy` produces the following:

`Argument 1 to "fun" has incompatible type "List[Tuple[int, int]]"; expected "List[Tuple[int, ...]]"`

**VS Code extension or command-line**
pyright extension v1.1.300

**Comment by kmillikin** [comment]
Mypy does seem to implement that `tuple[int, int]` is actually a subtype of `tuple[int, ...]`.  But pyright and pytype definitely don't and apparently neither does pyre (anecdotally, I haven't tested it).

Those other checkers implement that the two types are "consistent".  Then since `list` is invariant, the code above would be expected to type check given that interpretation.

I would probably wait until the typing community makes some decision one way or the other to avoid churn.

**Comment by erictraut** [comment]
Thanks. Closing this for now. Will revisit in the future when consensus is reached on this topic.
